### PR TITLE
[IOTDB-1463] Implement builder pattern for Session and SessionPool

### PR DESCRIPTION
**Pull request by liujiajun**
Providing a more fluent way to construct Session and SessionPool

### CODE REVIEW

File: session/src/test/java/org/apache/iotdb/session/SessionUT.java [hunk 1]
```diff
@@ -43,6 +43,7 @@
 
 import static org.junit.Assert.assertArrayEquals;
 import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
 import static org.junit.Assert.fail;
 
 public class SessionUT {
```
Comment (wangchao316) [inline session/src/test/java/org/apache/iotdb/session/SessionUT.java:49]: generate, Generally, UT ends with Test,
IT ends with IT.

File: session/src/test/java/org/apache/iotdb/session/SessionUT.java [hunk 2]
```diff
@@ -244,4 +245,25 @@ public void createDeviceTemplate() throws IoTDBConnectionException, StatementExe
         "template1", schemaNames, measurementList, dataTypeList, encodingList, compressionTypes);
     session.setSchemaTemplate("template1", "root.sg.1");
   }
+
+  @Test
+  public void testBuilder() {
+    Session.Builder builder = new Session.Builder("localhost", 1234);
```
Comment (wangchao316) [inline session/src/test/java/org/apache/iotdb/session/SessionUT.java:251]: hi
I found this implementation to be inconsistent with the use cases on jira.
We need to return a session through sessionbuild.
Session session = new SessionBuilder().build()

**Review by wangchao316 — CHANGES_REQUESTED** [review]

File: session/src/test/java/org/apache/iotdb/session/SessionUT.java (hunk 2 above)
Comment (liujiajun) [reply session/src/test/java/org/apache/iotdb/session/SessionUT.java]: Hi, thanks for the review! I think the implementation is consistent (see https://dzone.com/articles/design-patterns-the-builder-pattern). When using the pattern, however, we usually do not keep the interim builder object. I have changed the test case to make this more explicit.

Comment (wangchao316) [reply session/src/test/java/org/apache/iotdb/session/SessionUT.java]: default of constructors is no parameter.
// default
Session session = new SessionBuilder().build()
SessionPool sessionpool = new SessionPoolBuilder().build()

Comment (liujiajun) [reply session/src/test/java/org/apache/iotdb/session/SessionUT.java]: Correct me if I am wrong: `host` and `rpcPort` are compulsory fields for session and has no default values, so they are placed in the constructor. Same goes for sessionpool. If they are not put in constructor, users might forget to set values for them.

**Review by LebronAl — COMMENTED** [review]
Great Job！
Only minor issues!
1. As we can directly use `sh sbin/start-cli.sh` to connect server with default host "127.0.0.1" and default port "6667" using default user "root" with default password "root", So I think the host and port of session/sessionPool can also have default values, so that on the one hand, it is more convenient to test IoTDB locally, on the other hand, it makes the session interface consistent with CLI. BTW, it's my fault to make a wrong example. Actually, A built-in-class builder creation must be performed just like this "Session.Builder.build()" rather than 'SessionBuilder.build()'
2. Please update the docs.
3. After this PR has been merged, please send an email to the community telling everyone how the new session/sessionPool can be created now.

**Comment by liujiajun** [comment]
Thanks for the comments! Have changed accordingly. 

I removed the old construction method in the docs and also changed examples, so that new users can use builder as a preferred way.

Will blast out an email after this is merged~

File: session/src/test/java/org/apache/iotdb/session/SessionUT.java (hunk 2 above)
Comment (wangchao316) [reply session/src/test/java/org/apache/iotdb/session/SessionUT.java]: ok

File: session/src/main/java/org/apache/iotdb/session/pool/SessionPool.java [hunk 3]
```diff
@@ -1161,4 +1161,121 @@ public SessionDataSetWrapper executeRawDataQuery(List<String> paths, long startT
...
+
+  public boolean isEnableCompression() {
+    return enableCompression;
+  }
+
+  public boolean isEnableCacheLeader() {
+    return enableCacheLeader;
+  }
+
+  public ZoneId getZoneId() {
```
Comment (LebronAl) [inline session/src/main/java/org/apache/iotdb/session/pool/SessionPool.java:1201]: Currently, the setup of zondId needs a rpc `setTimeZone` which you can refer to [PR3457](https://github.com/apache/iotdb/pull/3457/files), so maybe we can not initialized zoneId here, what do you think?

**Review by LebronAl — COMMENTED** [review]
Last minor issue!

**Review by LebronAl — APPROVED** [review]
LGTM

**Review by wangchao316 — APPROVED** [review]
LGTM

**Pull request merged**